test/index.coffee

```diff
@@ -0,0 +1,82 @@
+Promise = require 'bluebird'
+{ expect } = require 'chai'
+_ = require 'lodash'
+
+gc = require('../lib/index')
+dockerUtils = require('../lib/docker.coffee')
+
+SKIP_GC_TEST = process.env.SKIP_GC_TEST == '1' || false
+
+promiseToBool = (p) ->
+	p.return(true).catchReturn(false)
+
+pullAsync = (docker, tag) ->
+	new Promise (resolve, reject) ->
+		docker.pull tag, (err, stream) ->
+			return reject(err) if err?
+			stream.resume()
+			stream.once('error', reject)
+			stream.once('end', resolve)
+
+dockerUtils.getDocker()
+.then (docker) ->
+
+	# This test case is a little weird, it requires that no other images are present on
+	# the system to ensure that the correct one is being removed. Because of this, you
+	# can use the SKIP_GC_TEST env var to inform the test suite not to run this test
+	describe 'Garbage collection', ->
+
+		it 'should remove the LRU image', ->
+			this.timeout(600000)
+			return Promise.resolve() if SKIP_GC_TEST
+			# first pull some images, so we know in which order they are referenced
+
+			Promise.map [
+				'alpine:3.1'
+				'debian:squeeze'
+				'ubuntu:lucid'
+			], (image) -> pullAsync(docker, image)
+			.then ->
+				# Attempt to remove a single byte, which will remove the LRU image,
+				# which should be alpine
+				gc.garbageCollect(1)
+			.then ->
+				# Check if the alpine image has been removed
+				docker.getImage('alpine:3.1')
+				.inspect()
+				.then ->
+					Promise.reject('Alpine image still present!')
```

Comment:
You're rejecting with a string - this should at the very least be an Error.  Beyond that you can do:
```coffee
.throw(new Error('Alpine image still present!'))